check-http/lib/check_http.go

```diff
@@ -245,22 +279,34 @@ func Run(args []string) *checkers.Checker {
 		}
 		tr.DialContext = newReplacableDial(dialer, resolves)
 	}
-	client := &http.Client{Transport: tr}
+	client := &http.Client{
+		Transport: tr,
+		Timeout:   time.Second * time.Duration(opts.Timeout),
+	}
 	client.CheckRedirect = func(req *http.Request, via []*http.Request) error {
 		if len(via) > opts.MaxRedirects {
 			return http.ErrUseLastResponse
 		}
 		return nil
 	}
 
-	req, err := http.NewRequest(http.MethodGet, opts.URL, nil)
-	if err != nil {
-		return checkers.Unknown(err.Error())
+	var req *http.Request
+	if opts.Body == "" {
+		req, err = http.NewRequest(opts.Method, opts.URL, nil)
+		if err != nil {
+			return checkers.Unknown(err.Error())
+		}
+	} else {
+		buffer := bytes.NewBuffer([]byte(opts.Body))
+		req, err = http.NewRequest(opts.Method, opts.URL, buffer)
```

Comment:
Is this better to use `strings.NewReader`?
https://golang.org/pkg/strings/#NewReader